src/types/index.ts

```diff
@@ -512,10 +512,16 @@ export class StarknetContract {
         }
 
         const adaptedInput = this.adaptInput(functionName, args);
+
+        // omit cairo 1.0 ABIs as it's not supported by the cairo-lang parser.
+        const abi =
+            this.abiPath && !fs.readFileSync(this.abiPath, "utf8").includes("core::")
+                ? this.abiPath
+                : undefined;
```

Comment:
`InteractWrapperOptions` should be edited to have `abi` as an optional property (`?`). Also, shouldn't then `--abi` be optional passed in starknet-wrapper?